### PR TITLE
ISPN-2939 Remove async marshalling

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-2939

### CODE REVIEW

File: core/src/test/java/org/infinispan/replication/ForceSyncAsyncFlagsTest.java [hunk 1]
```diff
@@ -61,34 +61,7 @@ public void testForceAsyncFlagUsage() throws Exception {
       // verify FORCE_ASYNCHRONOUS flag on SYNC cache
       cache1.withFlags(Flag.FORCE_ASYNCHRONOUS).put("k", "v");
       verify(mockTransport).invokeRemotely(anyCollectionOf(Address.class),
-                                           any(ReplicableCommand.class), eq(ResponseMode.ASYNCHRONOUS_WITH_SYNC_MARSHALLING), anyLong(),
-                                           any(ResponseFilter.class), any(DeliverOrder.class), anyBoolean());
-   }
-
-   public void testForceSyncFlagUsage() throws Exception {
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/replication/ForceSyncAsyncFlagsTest.java]: I think this test should stay, at least until we remove the FORCE_SYNCHRONOUS flag...

**Comment by danberindei** [comment]
Failure in ParserOverrideTest.testNamedCacheOverride 

    javax.xml.stream.XMLStreamException: ParseError at [row,col]:[7,143]
    Message: Unexpected attribute 'async-marshalling' encountered

There are also a couple instances of "async-marshalling" left in `sample.xml` in core and `named-async-cache` in the spring modules.

**Comment by tristantarrant** [comment]
Fixed the two suggestions. I also removed all references to async-marshalling in server (duh !)

**Comment by danberindei** [comment]
Integrated, thanks Tristan!

**Pull request closed without merging**